code/modules/species/station/xenomorph_hybrids/xeno_hybrids.dm

```diff
@@ -95,46 +94,69 @@
 		O_PLASMA =		/obj/item/organ/internal/xenos/plasmavessel/hunter,//Important for the xenomorph abilities, hunter to have a pretty small plasma capacity
 		O_STOMACH =		/obj/item/organ/internal/stomach,
 		O_INTESTINE =	/obj/item/organ/internal/intestine,
-		O_RESIN =		/obj/item/organ/internal/xenos/resinspinner,
+		O_RESIN =		/obj/item/organ/internal/xenos/resinspinner/hybrid,
 		)
 	vision_organ = O_BRAIN//Neomorphs have no (visible) Eyes, seeing without them should be possible.
 
 	reagent_tag = IS_XENOHYBRID
 
+	var/heal_rate = 0.5 //Lets just create a set number
+
 /datum/species/xenohybrid/can_breathe_water()
 	return TRUE	//they dont quite breathe
 
 /datum/species/xenohybrid/handle_environment_special(var/mob/living/carbon/human/H)
-	var/heal_amount = min(H.nutrition, 200) / 50 //Not to much else we might as well give them a diona like healing
-	H.nutrition = max(H.nutrition-heal_amount,0)
-
-	if(H.resting)
-		heal_amount *= 1.05//resting allows you to heal a little faster
-	var/fire_damage = H.getFireLoss()
-	if(fire_damage >= heal_amount)
-		H.adjustFireLoss(-heal_amount)
-		heal_amount = 0;
-		return
-	if(fire_damage < heal_amount)
-		H.adjustFireLoss(-heal_amount)
-		heal_amount -= fire_damage
-
-	var/trauma_damage = H.getBruteLoss()
-	if(trauma_damage >= heal_amount)
-		H.adjustBruteLoss(-heal_amount)
-		heal_amount = 0;
+	var/healing_factor = 1
+	if(H.lying)
+		healing_factor *= 1.2
+	if(H.active_regen)
+		if(!H.lying)
+			to_chat(H, SPAN_BOLDWARNING("You need to lie down to benefit from your enhanced regeneration"))
+			H.active_regen = FALSE
+		healing_factor *= 4
+	var/turf/T = get_turf(H)
+	if(/obj/effect/alien/weeds in T.contents)
+		healing_factor *= 1.1
+	if(/obj/structure/bed/hybrid_nest in T.contents)
+		healing_factor *= 1.2
+
+	if(H.nutrition < 50)
+		if(H.active_regen)
```

Comment:
you already have a check for active_regen a few lines before this where you could check the user's nutrition